storage/file.go

```diff
@@ -114,6 +132,24 @@ func (f *File) Create(maxSize uint64) error {
 	return nil
 }
 
+// CopyFile operation copied a file/blob from the sourceURL to the path provided.
+//
+// See https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/copy-file
+func (f File) CopyFile(sourceURL string, options *FileRequestOptions, metadata map[string]string) error {
+	extraHeaders := map[string]string{
+		"x-ms-type":        "file",
+		"x-ms-copy-source": sourceURL,
+	}
+
+	headers, err := f.fsc.createResource(f.buildPath(), resourceFile, options.getParameters(), mergeMDIntoExtraHeaders(f.Metadata, extraHeaders))
+	if err != nil {
+		return err
+	}
+
+	f.updateEtagAndLastModified(headers)
+	return nil
```

Comment:
Returning x-ms-copy-id.... but you listed that twice. Was there a second you were thinking of?   Status?